src/Makefile.am

```diff
@@ -1,7 +1,7 @@
 ## Process this file with autoconf to produce configure.
 
-AM_CFLAGS   = -Wall -Wwrite-strings $(PTHREAD_CFLAGS)
-AM_CXXFLAGS = -Wall -Wwrite-strings -Woverloaded-virtual $(PTHREAD_CFLAGS)
+AM_CFLAGS   = -Wall -Wwrite-strings $(PTHREAD_CFLAGS) -fgnu-tm
```

Comment:
Oh, my. That could be a challenge. Any suggestions?
It doesn't even look like there's a reliable compiler define:
$ gcc -dM -E - <<<'' | sort > gcc.out; gcc -fgnu-tm -dM -E - <<<'' | sort > gcctm.out; diff gcc.out gcctm.out
212a213
> #define _REENTRANT 1